Mathlib/CategoryTheory/Bicategory/End.lean

```diff
@@ -41,6 +41,7 @@ instance (X : C) : MonoidalCategory (EndMonoidal X) where
   tensorObj f g := f ≫ g
   whiskerLeft {f g h} η := f ◁ η
   whiskerRight {f g} η h := η ▷ h
+  tensorHom {X₁ Y₁ X₂ Y₂} f g := (f ▷ X₂) ≫ (Y₁ ◁ g)
```

Comment:
I think this is the only place that actually benefitted from the old constructor, and obvious the benefit was pretty marginal!

Reply:
With the previous API, there was not even a chance to use the old constructor! I would suggest not merging this PR, as it goes in the other direction as @yuma-mizuno's planned refactor.

Reply:
I'm in no rush to get this in, so am happy for it to sit around until @yuma-mizuno gets back to their refactor. My impression though is that `tensorHom` is the operation we actually have available in (almost) all the concrete cases, and is therefore the more convenient constructor; and was that @yuma-mizuno's refactor is motivated by `tensorHom` being a bad simp-normal form, which is possibly orthogonal to it being a bad constructor argument.